aesara/tensor/basic.py

```diff
@@ -978,7 +978,9 @@ def zeros_like(model, dtype=None, opt=False):
 
 def zeros(shape, dtype=None):
     """Create a `TensorVariable` filled with zeros, closer to NumPy's syntax than ``alloc``."""
-    if not isinstance(shape, (np.ndarray, Sequence, TensorVariable)):
+    if (not isinstance(shape, (np.ndarray, Sequence, TensorVariable))) or (
+        isinstance(shape, TensorVariable) and len(shape.shape.eval()) == 0
```

Comment:
Using `eval()` is a very costly and unsafe operation (e.g., may require inputs to be given)

Something like this might work instead:

```suggestion
        isinstance(shape, TensorVariable) and shape.ndim == 0
```